### PR TITLE
Fix versioning modification issues

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java [hunk 1]
```diff
@@ -98,6 +99,31 @@ public void addProperty(Session session, IdentificationType id, String key, Stri
...
+        Transaction transaction = session.beginTransaction();
+        session.saveOrUpdate(new MetadataPropertyMappingEntity(id0, value));
+        transaction.commit();
+      } else {
+        throw new ModelDBException("Entity not found for given key: " + key, Code.NOT_FOUND);
+      }
+      return true;
+    } catch (Exception ex) {
+      if (ModelDBUtils.needToRetry(ex)) {
+        return addProperty(id, key, value);
```
Comment (evgeniilezhnin) [inline backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: should be updateProperty?

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java [hunk 2]
```diff
@@ -98,6 +99,31 @@ public void addProperty(Session session, IdentificationType id, String key, Stri
     session.saveOrUpdate(new MetadataPropertyMappingEntity(id0, value));
   }
 
+  @Override
+  public boolean updateProperty(IdentificationType id, String key, String value)
```
Comment (evgeniilezhnin) [inline backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: is it used anywhere?

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: No. I am working on this PR so I will remove it.

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java [hunk 3]
```diff
@@ -185,11 +211,13 @@ public boolean deleteLabels(IdentificationType id, List<String> labels, boolean
             .append(ModelDBConstants.ID)
```
Comment (evgeniilezhnin) [inline backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java:185]: deleteLabels(
      Session session, IdentificationType id, List<String> labels, boolean deleteAll) seem to be not used anymore too

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: Method removed

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java]: Method removed

File: backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java (hunk 3 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/metadata/MetadataDAORdbImpl.java:185]: Method removed

File: backend/src/main/java/ai/verta/modeldb/dataset/DatasetServiceImpl.java [hunk 4]
```diff
@@ -563,7 +561,7 @@ public void getDatasetTags(GetTags request, StreamObserver<GetTags.Response> res
     try (RequestLatencyResource latencyResource =
         new RequestLatencyResource(ModelDBAuthInterceptor.METHOD_NAME.get())) {
       // Request Parameter Validation
-      throw new ModelDBException("Not supported", io.grpc.Status.Code.FAILED_PRECONDITION);
+      throw new ModelDBException("Not supported", Code.UNIMPLEMENTED);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/dataset/DatasetServiceImpl.java:565]: add a log statement

File: backend/src/main/resources/liquibase/create-versioning-tables-changelog.xml [hunk 5]
```diff
@@ -235,4 +235,15 @@
...
+        <preConditions onFail="MARK_RAN">
+            <not>
+                <columnExists tableName="labels_mapping" columnName="label_index"/>
+            </not>
+        </preConditions>
+        <addColumn tableName="labels_mapping">
+            <column name="label_index" type="int4" defaultValue="0"/>
+        </addColumn>
+    </changeSet>
+
```
Comment (ravishetye) [inline backend/src/main/resources/liquibase/create-versioning-tables-changelog.xml]: please revert, lets sort by label to get a consistent ordering

**Review by evgeniilezhnin — APPROVED** [review]

File: backend/src/main/java/ai/verta/modeldb/dataset/DatasetServiceImpl.java (hunk 4 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/dataset/DatasetServiceImpl.java:565]: Added Info log

File: backend/src/main/resources/liquibase/create-versioning-tables-changelog.xml (hunk 5 above)
Comment (anandjakhaniya) [reply backend/src/main/resources/liquibase/create-versioning-tables-changelog.xml]: Removed and modify all tag relevant test cases by alphanumeric order

**Review by ravishetye — APPROVED** [review]

**Pull request merged**